src/terminal/parser/OutputStateMachineEngine.hpp

```diff
@@ -229,9 +229,9 @@ namespace Microsoft::Console::VirtualTerminal
 
         static bool s_HexToUint(const wchar_t wch,
                                 unsigned int& value) noexcept;
-        bool _GetOscSetColorTable(const std::wstring_view string,
+        static bool s_GetOscSetColorTable(const std::wstring_view string,

```

Comment:
So...I found that it is perfectly OK for `_GetOscSetColorTable` to be static. So I changed it. Is it OK? Fact that other methods can also be static. I believe static methods actually reduces footprint of the code.

Reply:
Furthermore, I'm starting to feel that all the OSC parsing thing should be handled in a separate class `OscParser` or something. In my not-finished sixel implementation branch I created a `SixelParser` hpp and cpp to just for sixel parsing because it's too much logic to sqeeze in `OutputStateMachineEngine`. I'm beginning to feel OSC is also getting towards that direction.

Reply:
I think maybe this needs a bit of spec agreed on before you spend too much time on it. My assumption was that the parsing for the DCS command string would be handled in the dispatch classes - I don't think the state machine should need to know about all the different sequence formats. Once it's identified the operation from the final character, I think it should be streaming the command string straight through to the adapter so it can be processed immediately as the data arrives.

In the case of Sixel graphics, this would allow the adapter to render the image straight to the screen as the pixels are received. I think there are other DCS sequences, like music operations, that would have similar requirements. And yes, OSC might fall into this category too.

For the majority of simple operations that don't require streaming, though, there could be a default handler that just gathers everything up in a string, and passes that along once the string terminator is received.

Anyway, I don't want to sidetrack this PR too much with this discussion, but I do think we need to spend some time thinking through the DCS requirements before we lock down the architecture.

Reply:
I agree. I don’t want to make this PR bloated, either. And I’m still trying to figure out the best way to do the future work. Just showing some of my initial ideas.

Reply:
On second thought, I restored this and added public API tests for this. So adding multi param OSC 4 will be easier.